app/views/compute_resources_vms/form/libvirt/_base.html.erb

```diff
@@ -8,6 +8,13 @@
 <% checked = params[:host] && params[:host][:compute_attributes] && params[:host][:compute_attributes][:start] || '1' %>
 <%= checkbox_f f, :start, { :checked => (checked == '1'), :help_inline => _("Power ON this machine"), :label => _('Start'), :label_size => "col-md-2"} if new_vm && controller_name != "compute_attributes" %>
 
+<% firmware_type = new_vm && controller_name != 'compute_attributes' ? 'automatic' : compute_resource.firmware_type(f.object.firmware, f.object.secure_boot) %>
```

Comment:
I've decided to remove the conditions so that `firmware_type` is now consistently determined by the `compute_resource.firmware_type` method. The previous logic aimed to set the firmware to `Automatic` instead of `BIOS` for new VMs, but it didn't account for scenarios where the firmware is inherited from the Compute Profile.
```suggestion
<% firmware_type = compute_resource.firmware_type(f.object.firmware, f.object.secure_boot) %>
```